### DISCUSSION TITLE
parcel rust wasm and tauri

### DISCUSSION THREAD

**Discussion by tcastelly** [tauri-apps/tauri#1294] [Q&A] [answered]
Hello,

Thank you for this amazing project!

I noticed that it's possible by serving a project with parcel to use out of the box Rust. I'm wondering if it's possible to use both, Tauri and Parcel.

**Comment by nklayman** [comment, marked as answer]
Parcel can't be used to integrate Tauri, as Tauri can't be built to WASM, it has to be built to native code with the Tauri CLI. However, you can use a parcel-built website with Tauri just fine.

**Reply by tcastelly** [reply]
Thank you for your answer.

I saw in the roadmap, wasm should be available. Does that mean it will be possible to consume wasm modules with Tauri?
My idea would be to use VueJS + Rust (wasm), Tauri for system tasks and build to native code.

**Reply by nklayman** [reply]
Currently using wasm in the webview is supported (just like in a normal website). The roadmap item was for allowing Tauri commands to be called in a browser through wasm, so your Tauri commands would work inside of a browser outside of a desktop app too. However, after discussing with the rest of the team, we decided this wouldn't be practical and will be removing it from our raodmap. Thanks for your interest in Tauri, hopefully this makes things clear for you (and other future visitors) :).

**Reply by SOVLOOKUP** [reply]
@nklayman so using pure wasm in the webview is supported, but wasm with tauri commands not?

**Reply by nklayman** [reply]
Exactly. You can use regular WASM in a Tauri app, but Tauri commands can only be compiled to native platforms and used inside a Tauri app.

**Reply by OJFord** [reply]
Just to check I understand @nklayman - WASM as a target for Tauri, so that a Tauri app can be compiled cross-platform native & browser, (or just to target the browser at all) is not possible and not (any longer) planned?
